YARN-10541. capture the performance metrics of ZKRMStateStore

Neilxzn opened the pull request:
## detail 
In a production env, most yarn resourcemnagers are in ha mode and rely on zookeeper components. It is very necessary to capture the performance metrics of ZKRMStateStore.
https://issues.apache.org/jira/browse/YARN-10541

## test
add TestZKRMStateStore.testMetricsInited

Neilxzn committed:
add ZKRMStateStoreOpDurations & add testMetricsInited

Neilxzn commented:
@jojochuang @aajisaka What do you think about the patch? Thank you for your review.

hadoop-yetus commented:
:confetti_ball: **+1 overall**






| Vote | Subsystem | Runtime |  Logfile | Comment |
|:----:|----------:|--------:|:--------:|:-------:|
| +0 :ok: |  reexec  |   0m 35s |  |  Docker mode activated.  |
|||| _ Prechecks _ |
| +1 :green_heart: |  dupname  |   0m  0s |  |  No case conflicting files found.  |
| +1 :green_heart: |  @author  |   0m  0s |  |  The patch does not contain any @author tags.  |
| +1 :green_heart: |   |   0m  0s | [test4tests](test4tests) |  The patch appears to include 1 new or modified test files.  |
|||| _ trunk Compile Tests _ |
| +1 :green_heart: |  mvninstall  |  33m  6s |  |  trunk passed  |
| +1 :green_heart: |  compile  |   1m  3s |  |  trunk passed with JDK Ubuntu-11.0.9.1+1-Ubuntu-0ubuntu1.18.04  |
| +1 :green_heart: |  compile  |   0m 51s |  |  trunk passed with JDK Private Build-1.8.0_275-8u275-b01-0ubuntu1~18.04-b01  |
| +1 :green_heart: |  checkstyle  |   0m 40s |  |  trunk passed  |
| +1 :green_heart: |  mvnsite  |   0m 55s |  |  trunk passed  |
| +1 :green_heart: |  shadedclient  |  17m 19s |  |  branch has no errors when building and testing our client artifacts.  |
| +1 :green_heart: |  javadoc  |   0m 42s |  |  trunk passed with JDK Ubuntu-11.0.9.1+1-Ubuntu-0ubuntu1.18.04  |
| +1 :green_heart: |  javadoc  |   0m 37s |  |  trunk passed with JDK Private Build-1.8.0_275-8u275-b01-0ubuntu1~18.04-b01  |
| +0 :ok: |  spotbugs  |   1m 49s |  |  Used deprecated FindBugs config; considering switching to SpotBugs.  |
| +1 :green_heart: |  findbugs  |   1m 47s |  |  trunk passed  |
|||| _ Patch Compile Tests _ |
| +1 :green_heart: |  mvninstall  |   0m 54s |  |  the patch passed  |
| +1 :green_heart: |  compile  |   0m 56s |  |  the patch passed with JDK Ubuntu-11.0.9.1+1-Ubuntu-0ubuntu1.18.04  |
| +1 :green_heart: |  javac  |   0m 56s |  |  the patch passed  |
| +1 :green_heart: |  compile  |   0m 48s |  |  the patch passed with JDK Private Build-1.8.0_275-8u275-b01-0ubuntu1~18.04-b01  |
| +1 :green_heart: |  javac  |   0m 48s |  |  the patch passed  |
| -0 :warning: |  checkstyle  |   0m 33s | [/diff-checkstyle-hadoop-yarn-project_hadoop-yarn_hadoop-yarn-server_hadoop-yarn-server-resourcemanager.txt](https://ci-hadoop.apache.org/job/hadoop-multibranch/job/PR-2568/1/artifact/out/diff-checkstyle-hadoop-yarn-project_hadoop-yarn_hadoop-yarn-server_hadoop-yarn-server-resourcemanager.txt) |  hadoop-yarn-project/hadoop-yarn/hadoop-yarn-server/hadoop-yarn-server-resourcemanager: The patch generated 10 new + 33 unchanged - 0 fixed = 43 total (was 33)  |
| +1 :green_heart: |  mvnsite  |   0m 47s |  |  the patch passed  |
| +1 :green_heart: |  whitespace  |   0m  0s |  |  The patch has no whitespace issues.  |
| +1 :green_heart: |  shadedclient  |  15m 15s |  |  patch has no errors when building and testing our client artifacts.  |
| +1 :green_heart: |  javadoc  |   0m 37s |  |  the patch passed with JDK Ubuntu-11.0.9.1+1-Ubuntu-0ubuntu1.18.04  |
| +1 :green_heart: |  javadoc  |   0m 37s |  |  the patch passed with JDK Private Build-1.8.0_275-8u275-b01-0ubuntu1~18.04-b01  |
| +1 :green_heart: |  findbugs  |   1m 48s |  |  the patch passed  |
|||| _ Other Tests _ |
| +1 :green_heart: |  unit  |  89m 18s |  |  hadoop-yarn-server-resourcemanager in the patch passed.  |
| +1 :green_heart: |  asflicense  |   0m 33s |  |  The patch does not generate ASF License warnings.  |
|  |   | 171m 31s |  |  |


| Subsystem | Report/Notes |
|----------:|:-------------|
| Docker | ClientAPI=1.41 ServerAPI=1.41 base: https://ci-hadoop.apache.org/job/hadoop-multibranch/job/PR-2568/1/artifact/out/Dockerfile |
| GITHUB PR | https://github.com/apache/hadoop/pull/2568 |
| Optional Tests | dupname asflicense compile javac javadoc mvninstall mvnsite unit shadedclient findbugs checkstyle |
| uname | Linux ef6266fccb51 4.15.0-58-generic #64-Ubuntu SMP Tue Aug 6 11:12:41 UTC 2019 x86_64 x86_64 x86_64 GNU/Linux |
| Build tool | maven |
| Personality | dev-support/bin/hadoop.sh |
| git revision | trunk / a35fc3871b0 |
| Default Java | Private Build-1.8.0_275-8u275-b01-0ubuntu1~18.04-b01 |
| Multi-JDK versions | /usr/lib/jvm/java-11-openjdk-amd64:Ubuntu-11.0.9.1+1-Ubuntu-0ubuntu1.18.04 /usr/lib/jvm/java-8-openjdk-amd64:Private Build-1.8.0_275-8u275-b01-0ubuntu1~18.04-b01 |
|  Test Results | https://ci-hadoop.apache.org/job/hadoop-multibranch/job/PR-2568/1/testReport/ |
| Max. process+thread count | 883 (vs. ulimit of 5500) |
| modules | C: hadoop-yarn-project/hadoop-yarn/hadoop-yarn-server/hadoop-yarn-server-resourcemanager U: hadoop-yarn-project/hadoop-yarn/hadoop-yarn-server/hadoop-yarn-server-resourcemanager |
| Console output | https://ci-hadoop.apache.org/job/hadoop-multibranch/job/PR-2568/1/console |
| versions | git=2.17.1 maven=3.6.0 findbugs=4.0.6 |
| Powered by | Apache Yetus 0.13.0-SNAPSHOT https://yetus.apache.org |


This message was automatically generated.

jojochuang commented on hadoop-yarn-project/hadoop-yarn/hadoop-yarn-server/hadoop-yarn-server-resourcemanager/src/test/java/org/apache/hadoop/yarn/server/resourcemanager/recovery/TestZKRMStateStore.java:
```diff
@@ -1567,4 +1568,16 @@ public void testAppSubmissionContextIsPrunedInFinalApplicationState()
     store.close();
   }
+
+  @Test
```
Not familiar with YARN code. The production code looks fine to me. The test code should add more coverage for each of the methods in ZKRMStateStoreOpDurations.

Neilxzn replied:
done.  add more test in testMetricsInited to cover each of the methods in ZKRMStateStoreOpDurations.

Neilxzn committed:
add more test in  testMetricsInited

ZKRMStateStoreOpDurations.addLoadStateCallDuration
ZKRMStateStoreOpDurations.addStoreApplicationStateCallDuration
ZKRMStateStoreOpDurations.addUpdateApplicationStateCallDuration
ZKRMStateStoreOpDurations.addRemoveApplicationStateCallDuration

hadoop-yetus commented:
:broken_heart: **-1 overall**






| Vote | Subsystem | Runtime |  Logfile | Comment |
|:----:|----------:|--------:|:--------:|:-------:|
| +0 :ok: |  reexec  |   0m 33s |  |  Docker mode activated.  |
|||| _ Prechecks _ |
| +1 :green_heart: |  dupname  |   0m  0s |  |  No case conflicting files found.  |
| +1 :green_heart: |  @author  |   0m  0s |  |  The patch does not contain any @author tags.  |
| +1 :green_heart: |   |   0m  0s | [test4tests](test4tests) |  The patch appears to include 1 new or modified test files.  |
|||| _ trunk Compile Tests _ |
| +1 :green_heart: |  mvninstall  |  32m 46s |  |  trunk passed  |
| +1 :green_heart: |  compile  |   0m 59s |  |  trunk passed with JDK Ubuntu-11.0.9.1+1-Ubuntu-0ubuntu1.18.04  |
| +1 :green_heart: |  compile  |   0m 52s |  |  trunk passed with JDK Private Build-1.8.0_275-8u275-b01-0ubuntu1~18.04-b01  |
| +1 :green_heart: |  checkstyle  |   0m 42s |  |  trunk passed  |
| +1 :green_heart: |  mvnsite  |   0m 57s |  |  trunk passed  |
| +1 :green_heart: |  shadedclient  |  17m  4s |  |  branch has no errors when building and testing our client artifacts.  |
| +1 :green_heart: |  javadoc  |   0m 44s |  |  trunk passed with JDK Ubuntu-11.0.9.1+1-Ubuntu-0ubuntu1.18.04  |
| +1 :green_heart: |  javadoc  |   0m 39s |  |  trunk passed with JDK Private Build-1.8.0_275-8u275-b01-0ubuntu1~18.04-b01  |
| +0 :ok: |  spotbugs  |   1m 46s |  |  Used deprecated FindBugs config; considering switching to SpotBugs.  |
| +1 :green_heart: |  findbugs  |   1m 44s |  |  trunk passed  |
|||| _ Patch Compile Tests _ |
| +1 :green_heart: |  mvninstall  |   0m 50s |  |  the patch passed  |
| +1 :green_heart: |  compile  |   0m 52s |  |  the patch passed with JDK Ubuntu-11.0.9.1+1-Ubuntu-0ubuntu1.18.04  |
| +1 :green_heart: |  javac  |   0m 52s |  |  the patch passed  |
| +1 :green_heart: |  compile  |   0m 44s |  |  the patch passed with JDK Private Build-1.8.0_275-8u275-b01-0ubuntu1~18.04-b01  |
| +1 :green_heart: |  javac  |   0m 44s |  |  the patch passed  |
| -0 :warning: |  checkstyle  |   0m 32s | [/diff-checkstyle-hadoop-yarn-project_hadoop-yarn_hadoop-yarn-server_hadoop-yarn-server-resourcemanager.txt](https://ci-hadoop.apache.org/job/hadoop-multibranch/job/PR-2568/2/artifact/out/diff-checkstyle-hadoop-yarn-project_hadoop-yarn_hadoop-yarn-server_hadoop-yarn-server-resourcemanager.txt) |  hadoop-yarn-project/hadoop-yarn/hadoop-yarn-server/hadoop-yarn-server-resourcemanager: The patch generated 15 new + 33 unchanged - 0 fixed = 48 total (was 33)  |
| +1 :green_heart: |  mvnsite  |   0m 47s |  |  the patch passed  |
| +1 :green_heart: |  whitespace  |   0m  0s |  |  The patch has no whitespace issues.  |
| +1 :green_heart: |  shadedclient  |  14m 48s |  |  patch has no errors when building and testing our client artifacts.  |
| +1 :green_heart: |  javadoc  |   0m 38s |  |  the patch passed with JDK Ubuntu-11.0.9.1+1-Ubuntu-0ubuntu1.18.04  |
| +1 :green_heart: |  javadoc  |   0m 34s |  |  the patch passed with JDK Private Build-1.8.0_275-8u275-b01-0ubuntu1~18.04-b01  |
| +1 :green_heart: |  findbugs  |   1m 47s |  |  the patch passed  |
|||| _ Other Tests _ |
| -1 :x: |  unit  |  89m 40s | [/patch-unit-hadoop-yarn-project_hadoop-yarn_hadoop-yarn-server_hadoop-yarn-server-resourcemanager.txt](https://ci-hadoop.apache.org/job/hadoop-multibranch/job/PR-2568/2/artifact/out/patch-unit-hadoop-yarn-project_hadoop-yarn_hadoop-yarn-server_hadoop-yarn-server-resourcemanager.txt) |  hadoop-yarn-server-resourcemanager in the patch passed.  |
| +1 :green_heart: |  asflicense  |   0m 33s |  |  The patch does not generate ASF License warnings.  |
|  |   | 170m 11s |  |  |


| Reason | Tests |
|-------:|:------|
| Failed junit tests | hadoop.yarn.server.resourcemanager.security.TestDelegationTokenRenewer |


| Subsystem | Report/Notes |
|----------:|:-------------|
| Docker | ClientAPI=1.41 ServerAPI=1.41 base: https://ci-hadoop.apache.org/job/hadoop-multibranch/job/PR-2568/2/artifact/out/Dockerfile |
| GITHUB PR | https://github.com/apache/hadoop/pull/2568 |
| Optional Tests | dupname asflicense compile javac javadoc mvninstall mvnsite unit shadedclient findbugs checkstyle |
| uname | Linux 489d26c5a29d 4.15.0-58-generic #64-Ubuntu SMP Tue Aug 6 11:12:41 UTC 2019 x86_64 x86_64 x86_64 GNU/Linux |
| Build tool | maven |
| Personality | dev-support/bin/hadoop.sh |
| git revision | trunk / 16a20503cac |
| Default Java | Private Build-1.8.0_275-8u275-b01-0ubuntu1~18.04-b01 |
| Multi-JDK versions | /usr/lib/jvm/java-11-openjdk-amd64:Ubuntu-11.0.9.1+1-Ubuntu-0ubuntu1.18.04 /usr/lib/jvm/java-8-openjdk-amd64:Private Build-1.8.0_275-8u275-b01-0ubuntu1~18.04-b01 |
|  Test Results | https://ci-hadoop.apache.org/job/hadoop-multibranch/job/PR-2568/2/testReport/ |
| Max. process+thread count | 865 (vs. ulimit of 5500) |
| modules | C: hadoop-yarn-project/hadoop-yarn/hadoop-yarn-server/hadoop-yarn-server-resourcemanager U: hadoop-yarn-project/hadoop-yarn/hadoop-yarn-server/hadoop-yarn-server-resourcemanager |
| Console output | https://ci-hadoop.apache.org/job/hadoop-multibranch/job/PR-2568/2/console |
| versions | git=2.17.1 maven=3.6.0 findbugs=4.0.6 |
| Powered by | Apache Yetus 0.13.0-SNAPSHOT https://yetus.apache.org |


This message was automatically generated.

Neilxzn committed:
fix some checkstyle

hadoop-yetus commented:
:broken_heart: **-1 overall**






| Vote | Subsystem | Runtime |  Logfile | Comment |
|:----:|----------:|--------:|:--------:|:-------:|
| +0 :ok: |  reexec  |   0m 29s |  |  Docker mode activated.  |
|||| _ Prechecks _ |
| +1 :green_heart: |  dupname  |   0m  0s |  |  No case conflicting files found.  |
| +1 :green_heart: |  @author  |   0m  0s |  |  The patch does not contain any @author tags.  |
| +1 :green_heart: |   |   0m  0s | [test4tests](test4tests) |  The patch appears to include 1 new or modified test files.  |
|||| _ trunk Compile Tests _ |
| +1 :green_heart: |  mvninstall  |  32m 53s |  |  trunk passed  |
| +1 :green_heart: |  compile  |   1m  1s |  |  trunk passed with JDK Ubuntu-11.0.9.1+1-Ubuntu-0ubuntu1.18.04  |
| +1 :green_heart: |  compile  |   0m 52s |  |  trunk passed with JDK Private Build-1.8.0_275-8u275-b01-0ubuntu1~18.04-b01  |
| +1 :green_heart: |  checkstyle  |   0m 42s |  |  trunk passed  |
| +1 :green_heart: |  mvnsite  |   0m 56s |  |  trunk passed  |
| +1 :green_heart: |  shadedclient  |  17m 10s |  |  branch has no errors when building and testing our client artifacts.  |
| +1 :green_heart: |  javadoc  |   0m 44s |  |  trunk passed with JDK Ubuntu-11.0.9.1+1-Ubuntu-0ubuntu1.18.04  |
| +1 :green_heart: |  javadoc  |   0m 38s |  |  trunk passed with JDK Private Build-1.8.0_275-8u275-b01-0ubuntu1~18.04-b01  |
| +0 :ok: |  spotbugs  |   1m 47s |  |  Used deprecated FindBugs config; considering switching to SpotBugs.  |
| +1 :green_heart: |  findbugs  |   1m 43s |  |  trunk passed  |
|||| _ Patch Compile Tests _ |
| +1 :green_heart: |  mvninstall  |   0m 50s |  |  the patch passed  |
| +1 :green_heart: |  compile  |   0m 52s |  |  the patch passed with JDK Ubuntu-11.0.9.1+1-Ubuntu-0ubuntu1.18.04  |
| +1 :green_heart: |  javac  |   0m 52s |  |  the patch passed  |
| +1 :green_heart: |  compile  |   0m 44s |  |  the patch passed with JDK Private Build-1.8.0_275-8u275-b01-0ubuntu1~18.04-b01  |
| +1 :green_heart: |  javac  |   0m 44s |  |  the patch passed  |
| -0 :warning: |  checkstyle  |   0m 33s | [/diff-checkstyle-hadoop-yarn-project_hadoop-yarn_hadoop-yarn-server_hadoop-yarn-server-resourcemanager.txt](https://ci-hadoop.apache.org/job/hadoop-multibranch/job/PR-2568/3/artifact/out/diff-checkstyle-hadoop-yarn-project_hadoop-yarn_hadoop-yarn-server_hadoop-yarn-server-resourcemanager.txt) |  hadoop-yarn-project/hadoop-yarn/hadoop-yarn-server/hadoop-yarn-server-resourcemanager: The patch generated 5 new + 33 unchanged - 0 fixed = 38 total (was 33)  |
| +1 :green_heart: |  mvnsite  |   0m 46s |  |  the patch passed  |
| +1 :green_heart: |  whitespace  |   0m  0s |  |  The patch has no whitespace issues.  |
| +1 :green_heart: |  shadedclient  |  14m 48s |  |  patch has no errors when building and testing our client artifacts.  |
| +1 :green_heart: |  javadoc  |   0m 37s |  |  the patch passed with JDK Ubuntu-11.0.9.1+1-Ubuntu-0ubuntu1.18.04  |
| +1 :green_heart: |  javadoc  |   0m 34s |  |  the patch passed with JDK Private Build-1.8.0_275-8u275-b01-0ubuntu1~18.04-b01  |
| +1 :green_heart: |  findbugs  |   1m 46s |  |  the patch passed  |
|||| _ Other Tests _ |
| -1 :x: |  unit  |  89m 20s | [/patch-unit-hadoop-yarn-project_hadoop-yarn_hadoop-yarn-server_hadoop-yarn-server-resourcemanager.txt](https://ci-hadoop.apache.org/job/hadoop-multibranch/job/PR-2568/3/artifact/out/patch-unit-hadoop-yarn-project_hadoop-yarn_hadoop-yarn-server_hadoop-yarn-server-resourcemanager.txt) |  hadoop-yarn-server-resourcemanager in the patch passed.  |
| +1 :green_heart: |  asflicense  |   0m 34s |  |  The patch does not generate ASF License warnings.  |
|  |   | 169m 58s |  |  |


| Reason | Tests |
|-------:|:------|
| Failed junit tests | hadoop.yarn.server.resourcemanager.scheduler.capacity.TestCapacitySchedulerAutoQueueCreation |
|   | hadoop.yarn.server.resourcemanager.security.TestDelegationTokenRenewer |


| Subsystem | Report/Notes |
|----------:|:-------------|
| Docker | ClientAPI=1.41 ServerAPI=1.41 base: https://ci-hadoop.apache.org/job/hadoop-multibranch/job/PR-2568/3/artifact/out/Dockerfile |
| GITHUB PR | https://github.com/apache/hadoop/pull/2568 |
| Optional Tests | dupname asflicense compile javac javadoc mvninstall mvnsite unit shadedclient findbugs checkstyle |
| uname | Linux 82d22fe1260f 4.15.0-58-generic #64-Ubuntu SMP Tue Aug 6 11:12:41 UTC 2019 x86_64 x86_64 x86_64 GNU/Linux |
| Build tool | maven |
| Personality | dev-support/bin/hadoop.sh |
| git revision | trunk / 16a20503cac |
| Default Java | Private Build-1.8.0_275-8u275-b01-0ubuntu1~18.04-b01 |
| Multi-JDK versions | /usr/lib/jvm/java-11-openjdk-amd64:Ubuntu-11.0.9.1+1-Ubuntu-0ubuntu1.18.04 /usr/lib/jvm/java-8-openjdk-amd64:Private Build-1.8.0_275-8u275-b01-0ubuntu1~18.04-b01 |
|  Test Results | https://ci-hadoop.apache.org/job/hadoop-multibranch/job/PR-2568/3/testReport/ |
| Max. process+thread count | 878 (vs. ulimit of 5500) |
| modules | C: hadoop-yarn-project/hadoop-yarn/hadoop-yarn-server/hadoop-yarn-server-resourcemanager U: hadoop-yarn-project/hadoop-yarn/hadoop-yarn-server/hadoop-yarn-server-resourcemanager |
| Console output | https://ci-hadoop.apache.org/job/hadoop-multibranch/job/PR-2568/3/console |
| versions | git=2.17.1 maven=3.6.0 findbugs=4.0.6 |
| Powered by | Apache Yetus 0.13.0-SNAPSHOT https://yetus.apache.org |


This message was automatically generated.

Neilxzn commented:
The failed unit tests are  hadoop.yarn.server.resourcemanager.scheduler.capacity.TestCapacitySchedulerAutoQueueCreation and hadoop.yarn.server.resourcemanager.security.TestDelegationTokenRenewer.  These tests are not in the patch.  Please check it.

Neilxzn commented:
> The failed unit tests are hadoop.yarn.server.resourcemanager.scheduler.capacity.TestCapacitySchedulerAutoQueueCreation and hadoop.yarn.server.resourcemanager.security.TestDelegationTokenRenewer. These tests are not in the patch. Please check it.

@jojochuang Can you help me trigger jerkins to build it again?

jojochuang reviewed (approved):
Triggered a rebuild +1. PR looks good to me.

hadoop-yetus commented:
:broken_heart: **-1 overall**






| Vote | Subsystem | Runtime |  Logfile | Comment |
|:----:|----------:|--------:|:--------:|:-------:|
| +0 :ok: |  reexec  |  26m 15s |  |  Docker mode activated.  |
|||| _ Prechecks _ |
| +1 :green_heart: |  dupname  |   0m  0s |  |  No case conflicting files found.  |
| +1 :green_heart: |  @author  |   0m  0s |  |  The patch does not contain any @author tags.  |
| +1 :green_heart: |   |   0m  0s | [test4tests](test4tests) |  The patch appears to include 1 new or modified test files.  |
|||| _ trunk Compile Tests _ |
| +1 :green_heart: |  mvninstall  |  32m 59s |  |  trunk passed  |
| +1 :green_heart: |  compile  |   0m 59s |  |  trunk passed with JDK Ubuntu-11.0.9.1+1-Ubuntu-0ubuntu1.18.04  |
| +1 :green_heart: |  compile  |   0m 51s |  |  trunk passed with JDK Private Build-1.8.0_275-8u275-b01-0ubuntu1~18.04-b01  |
| +1 :green_heart: |  checkstyle  |   0m 41s |  |  trunk passed  |
| +1 :green_heart: |  mvnsite  |   0m 57s |  |  trunk passed  |
| +1 :green_heart: |  shadedclient  |  16m 55s |  |  branch has no errors when building and testing our client artifacts.  |
| +1 :green_heart: |  javadoc  |   0m 43s |  |  trunk passed with JDK Ubuntu-11.0.9.1+1-Ubuntu-0ubuntu1.18.04  |
| +1 :green_heart: |  javadoc  |   0m 39s |  |  trunk passed with JDK Private Build-1.8.0_275-8u275-b01-0ubuntu1~18.04-b01  |
| +0 :ok: |  spotbugs  |   1m 47s |  |  Used deprecated FindBugs config; considering switching to SpotBugs.  |
| +1 :green_heart: |  findbugs  |   1m 44s |  |  trunk passed  |
|||| _ Patch Compile Tests _ |
| +1 :green_heart: |  mvninstall  |   0m 52s |  |  the patch passed  |
| +1 :green_heart: |  compile  |   0m 52s |  |  the patch passed with JDK Ubuntu-11.0.9.1+1-Ubuntu-0ubuntu1.18.04  |
| +1 :green_heart: |  javac  |   0m 52s |  |  the patch passed  |
| +1 :green_heart: |  compile  |   0m 43s |  |  the patch passed with JDK Private Build-1.8.0_275-8u275-b01-0ubuntu1~18.04-b01  |
| +1 :green_heart: |  javac  |   0m 43s |  |  the patch passed  |
| -0 :warning: |  checkstyle  |   0m 32s | [/diff-checkstyle-hadoop-yarn-project_hadoop-yarn_hadoop-yarn-server_hadoop-yarn-server-resourcemanager.txt](https://ci-hadoop.apache.org/job/hadoop-multibranch/job/PR-2568/4/artifact/out/diff-checkstyle-hadoop-yarn-project_hadoop-yarn_hadoop-yarn-server_hadoop-yarn-server-resourcemanager.txt) |  hadoop-yarn-project/hadoop-yarn/hadoop-yarn-server/hadoop-yarn-server-resourcemanager: The patch generated 5 new + 33 unchanged - 0 fixed = 38 total (was 33)  |
| +1 :green_heart: |  mvnsite  |   0m 46s |  |  the patch passed  |
| +1 :green_heart: |  whitespace  |   0m  0s |  |  The patch has no whitespace issues.  |
| +1 :green_heart: |  shadedclient  |  14m 47s |  |  patch has no errors when building and testing our client artifacts.  |
| +1 :green_heart: |  javadoc  |   0m 38s |  |  the patch passed with JDK Ubuntu-11.0.9.1+1-Ubuntu-0ubuntu1.18.04  |
| +1 :green_heart: |  javadoc  |   0m 34s |  |  the patch passed with JDK Private Build-1.8.0_275-8u275-b01-0ubuntu1~18.04-b01  |
| +1 :green_heart: |  findbugs  |   1m 50s |  |  the patch passed  |
|||| _ Other Tests _ |
| -1 :x: |  unit  |  89m  5s | [/patch-unit-hadoop-yarn-project_hadoop-yarn_hadoop-yarn-server_hadoop-yarn-server-resourcemanager.txt](https://ci-hadoop.apache.org/job/hadoop-multibranch/job/PR-2568/4/artifact/out/patch-unit-hadoop-yarn-project_hadoop-yarn_hadoop-yarn-server_hadoop-yarn-server-resourcemanager.txt) |  hadoop-yarn-server-resourcemanager in the patch passed.  |
| +1 :green_heart: |  asflicense  |   0m 33s |  |  The patch does not generate ASF License warnings.  |
|  |   | 195m 22s |  |  |


| Reason | Tests |
|-------:|:------|
| Failed junit tests | hadoop.yarn.server.resourcemanager.security.TestDelegationTokenRenewer |


| Subsystem | Report/Notes |
|----------:|:-------------|
| Docker | ClientAPI=1.41 ServerAPI=1.41 base: https://ci-hadoop.apache.org/job/hadoop-multibranch/job/PR-2568/4/artifact/out/Dockerfile |
| GITHUB PR | https://github.com/apache/hadoop/pull/2568 |
| Optional Tests | dupname asflicense compile javac javadoc mvninstall mvnsite unit shadedclient findbugs checkstyle |
| uname | Linux eedb7594da5a 4.15.0-58-generic #64-Ubuntu SMP Tue Aug 6 11:12:41 UTC 2019 x86_64 x86_64 x86_64 GNU/Linux |
| Build tool | maven |
| Personality | dev-support/bin/hadoop.sh |
| git revision | trunk / 513f1995adc |
| Default Java | Private Build-1.8.0_275-8u275-b01-0ubuntu1~18.04-b01 |
| Multi-JDK versions | /usr/lib/jvm/java-11-openjdk-amd64:Ubuntu-11.0.9.1+1-Ubuntu-0ubuntu1.18.04 /usr/lib/jvm/java-8-openjdk-amd64:Private Build-1.8.0_275-8u275-b01-0ubuntu1~18.04-b01 |
|  Test Results | https://ci-hadoop.apache.org/job/hadoop-multibranch/job/PR-2568/4/testReport/ |
| Max. process+thread count | 876 (vs. ulimit of 5500) |
| modules | C: hadoop-yarn-project/hadoop-yarn/hadoop-yarn-server/hadoop-yarn-server-resourcemanager U: hadoop-yarn-project/hadoop-yarn/hadoop-yarn-server/hadoop-yarn-server-resourcemanager |
| Console output | https://ci-hadoop.apache.org/job/hadoop-multibranch/job/PR-2568/4/console |
| versions | git=2.17.1 maven=3.6.0 findbugs=4.0.6 |
| Powered by | Apache Yetus 0.13.0-SNAPSHOT https://yetus.apache.org |


This message was automatically generated.

jojochuang commented:
TestDelegationTokenRenewer#testRMRestartWithExpiredToken failed 3 times consecutively in the PR's tests, but doesn't reproduce in my local tree.  Probably very unlucky.

+1 from me.
